user/views.py

```diff
@@ -1,17 +1,55 @@
 from django.shortcuts import render, redirect
+from django.contrib.auth.hashers import make_password
+from django.contrib.auth import authenticate, login, logout
+from django.contrib import messages
+from .models import User
 
-# Create your views here.
 
 
 
+def home(request):
+    if request.user.is_authenticated:
+        return render(request,'tweet/home.html')
+    return render(request,'tweet/login.html')
 
-def check_login(request):
+def signup(request):
     if request.user.is_authenticated:
         return redirect('home')
-    return render(request,'tweet/index.html')
+    if request.method == 'GET':
+        return render(request,'tweet/register.html')
+    email = request.POST.get('email')
```

Comment:
Add `username=email.split('@').first`

Reply:
if we do so then in login we also need to do this so to use inbuilt authenticate function. please confirm.